### PR TITLE
iox-#1394 Address AUTOSAR warnings in `RelativePointerData`

### PR DESCRIPTION
**Pull request by mossmaurice**
Signed-off-by: Simon Hoinkis <simon.hoinkis@apex.ai>

## Pre-Review Checklist for the PR Author

1. [x] Code follows the coding style of [CONTRIBUTING.md][contributing]
1. [x] Tests follow the [best practice for testing][testing]
1. [x] ~~Changelog updated [in the unreleased section][changelog] including API breaking changes~~
1. [x] Branch follows the naming format (`iox-123-this-is-a-branch`)
1. [x] Commits messages are according to this [guideline][commit-guidelines]
    - [x] Commit messages have the issue ID (`iox-#123 commit text`)
    - [x] Commit messages are signed (`git commit -s`)
    - [x] Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
1. [x] Update the PR title
   - Follow the same conventions as for commit messages
   - Link to the relevant issue
1. [x] Relevant issues are linked
1. [x] Add sensible notes for the reviewer
1. [x] All checks have passed (except `task-list-completed`)
1. [x] All touched (C/C++) source code files are added to `./clang-tidy-diff-scans.txt`
1. [x] Assign PR to reviewer

[commit-guidelines]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[eca]: http://www.eclipse.org/legal/ECA.php
[contributing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/CONTRIBUTING.md#coding-style
[testing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/concepts/best-practice-for-testing.md
[changelog]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/release-notes/iceoryx-unreleased.md

## Notes for Reviewer

* Address AUTOSAR warnings in `RelativePointerData`

## Checklist for the PR Reviewer

- [x] Commits are properly organized and messages are according to the guideline
- [x] Code according to our coding style and naming conventions
- [x] Unit tests have been written for new behavior
    - [x] Each unit test case has a unique UUID
- [x] Public API changes are documented via doxygen
- [x] Copyright owner are updated in the changed files
- [x] All touched (C/C++) source code files have been added to `./clang-tidy-diff-scans.txt`
- [x] PR title describes the changes

## Post-review Checklist for the PR Author

1. [x] All open points are addressed and tracked via issues

## References

- Closes #1394 (partly)

### CODE REVIEW

File: iceoryx_hoofs/include/iceoryx_hoofs/internal/relocatable_pointer/relative_pointer_data.hpp [hunk 1]
```diff
@@ -69,13 +69,14 @@ class RelativePointerData
     /// @note the maximum offset which can be represented
     static constexpr offset_t MAX_VALID_OFFSET{OFFSET_RANGE - 1U};
     /// @note the maximum allowed size of RelativePointerData
+    // AXIVION Next Line AutosarC++19_03-M11.0.1 : Needs to be verified outside of the class
```
Comment (FerdinandSpitzschnueffler) [inline iceoryx_hoofs/include/iceoryx_hoofs/internal/relocatable_pointer/relative_pointer_data.hpp:72]: I wonder why this rule must not be suppressed in the lines above.

Comment (elBoberido) [reply iceoryx_hoofs/include/iceoryx_hoofs/internal/relocatable_pointer/relative_pointer_data.hpp:72]: Me too. It seems to not be required for the other constexpr

Comment (mossmaurice) [reply iceoryx_hoofs/include/iceoryx_hoofs/internal/relocatable_pointer/relative_pointer_data.hpp:72]: Correct. Turns out I was looking at a delta scan only :bulb:

File: iceoryx_hoofs/include/iceoryx_hoofs/internal/relocatable_pointer/relative_pointer_data.hpp [hunk 2]
```diff
@@ -30,7 +30,7 @@ namespace rp
 class RelativePointerData
 {
   public:
-    using id_t = uint16_t;
+    using identifier_t = uint16_t;
```
Comment (elBoberido) [inline iceoryx_hoofs/include/iceoryx_hoofs/internal/relocatable_pointer/relative_pointer_data.hpp:33]: This is the same `id_t` as in `relative_pointer.hpp` but it should not include this file if possible

Comment (mossmaurice) [reply iceoryx_hoofs/include/iceoryx_hoofs/internal/relocatable_pointer/relative_pointer_data.hpp:33]: @elBoberido The unification of the `identifier_t` / `id_t` types and naming will be done on a separate branch once this one and #1521 is merged.

**Review by dkroenke — APPROVED** [review]

File: iceoryx_hoofs/include/iceoryx_hoofs/internal/relocatable_pointer/relative_pointer_data.hpp [hunk 3]
```diff
@@ -57,25 +57,32 @@ class RelativePointerData
     bool isLogicalNullptr() const noexcept;
 
     /// @note the maximum number of available ids
-    static constexpr id_t ID_RANGE{std::numeric_limits<id_t>::max()};
+    // AXIVION Next Line AutosarC++19_03-M11.0.1 : Needs to be verified outside of the class
```
Comment (elBoberido) [inline iceoryx_hoofs/include/iceoryx_hoofs/internal/relocatable_pointer/relative_pointer_data.hpp:60]: I would actually vote to deactivate this rule else we have 100 suppressions in hoofs alone.

**Review by elBoberido — APPROVED** [review]

**Review by dkroenke — APPROVED** [review]

**Pull request merged**